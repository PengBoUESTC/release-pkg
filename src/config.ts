import { cwd } from 'process';
import load from '@proload/core';
import type { Config as Pre_Config } from '@proload/core';

function preloadPlugin() {
  return {
    name: '@proload/plugin-tsm',
    extensions: ['ts', 'tsx', 'cts', 'mts'],
    async register(fileName: string) {
      if (/\.([cm]ts|tsx?)$/.test(fileName)) {
        await require('tsm');
      }
    },
  };
}

load.use([preloadPlugin()]);

const DefaultConfig = {
  releaseBranch: ['master', 'main'],
  scripts: {
    build: 'build',
    changelog: 'changelog',
  },
  tag: true,
};

export interface Config {
  releaseBranch: string[];
  releaseUser?: string[];
  scripts: {
    build: string;
    changelog: string;
  };
  tag?: boolean;
  release?: boolean;
}

export const loadConfig = async (namespace: string): Promise<Config> => {
  const config = (await load(namespace, { cwd: cwd() })) as
    | Pre_Config<Config>
    | undefined;

  return config?.value || DefaultConfig;
};
