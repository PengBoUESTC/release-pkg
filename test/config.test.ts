import { loadConfig } from '../dist/config';

test('test config parse', async () => {
  const config = await loadConfig('release');
  expect(config).toMatchSnapshot();
});