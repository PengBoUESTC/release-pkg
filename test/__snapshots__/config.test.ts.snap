// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test config parse 1`] = `
{
  "releaseBranch": [
    "master",
    "main",
  ],
  "releaseUser": [
    "pengbo-study",
  ],
  "scripts": {
    "build": "build",
    "changelog": "changelog",
  },
  "tag": true,
}
`;
